chia/types/blockchain_format/program.py

```diff
@@ -267,3 +281,35 @@ def _sexp_replace(sexp: T_CLVMStorage, to_sexp: Callable[[Any], T_Program], **kw
     new_r = _sexp_replace(pair[1], to_sexp, **args_by_prefix.get("r", {}))
 
     return to_sexp((new_f, new_r))
+
+
+def _run(prg: Union[SerializedProgram, Program], max_cost: int, flags: int, args: Any) -> tuple[int, Program]:
+    if isinstance(prg, SerializedProgram):
+        result = prg.run_rust(max_cost, flags, args)
+        return result[0], Program(result[1])  # type: ignore[arg-type]
+    elif isinstance(prg, Program):
+        return prg._run(max_cost, flags, args)
+    else:  # pragme: no cover
+        assert False
```

Comment:
```suggestion
    else:  # pragma: no cover
        raise Exception(f"unsupported type passed: {type(prg).__name__}")
```

Reply:
This doesn't seem right to me. In my mind, an assertion means "programmer error" whereas some `Exception` being raised means something failed that's not in the programmer's control. Even though both turn into exceptions, I think this is generally a useful distinction.